homeassistant/components/smlight/switch.py

```diff
@@ -0,0 +1,118 @@
+
+from .coordinator import SmDataUpdateCoordinator
+from .entity import SmEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmSwitchEntityDescription(SwitchEntityDescription):
+    """Class to describe a Switch entity."""
+
+    setting: Settings
+    state_fn: Callable[[Sensors], bool | None]
+
+
+SWITCHES: Final = [
+    SmSwitchEntityDescription(
+        key="disable_led",
+        translation_key="disable_led",
+        setting=Settings.DISABLE_LEDS,
+        state_fn=lambda x: x.disable_leds,
+    ),
+    SmSwitchEntityDescription(
+        key="night_mode",
+        translation_key="night_mode",
+        setting=Settings.NIGHT_MODE,
+        state_fn=lambda x: x.night_mode,
+    ),
+    SmSwitchEntityDescription(
+        key="auto_zigbee_update",
+        translation_key="auto_zigbee_update",
+        entity_category=EntityCategory.CONFIG,
+        setting=Settings.ZB_AUTOUPDATE,
+        state_fn=lambda x: x.auto_zigbee,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize switches for SLZB-06 device."""
+    coordinator = entry.runtime_data
+
+    async_add_entities(SmSwitch(coordinator, switch) for switch in SWITCHES)
+
+
+class SmSwitch(SmEntity, SwitchEntity):
+    """Representation of a SLZB-06 switch."""
+
+    entity_description: SmSwitchEntityDescription
+    _attr_device_class = SwitchDeviceClass.SWITCH
+
+    def __init__(
+        self,
+        coordinator: SmDataUpdateCoordinator,
+        description: SmSwitchEntityDescription,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.unique_id}-{description.key}"
+
+        self._page, self._toggle = self.entity_description.setting.value
+        self._update_attrs()
+
+    async def set_smlight(self, state: bool):
+        """Set the state on SLZB device."""
+        await self.coordinator.client.set_toggle(self._page, self._toggle, state)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        self._attr_is_on = True
+        self.async_write_ha_state()
+
+        await self.set_smlight(True)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        self._attr_is_on = False
+        self.async_write_ha_state()
+
+        await self.set_smlight(False)
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle update from data coordinator."""
+        self._update_attrs()
+        super()._handle_coordinator_update()
+
+    def _update_attrs(self):
+        """Update the entity attributes."""
+        self._attr_is_on = self.entity_description.state_fn(
+            self.coordinator.data.sensors
+        )
```

Comment:
You can also just return this in a `is_on` property

Reply:
But the is_on is never called. The entity is made not to poll because that's the coordinator entity default. I would advice the SSE to work with the coordinator

Reply:
ok converted to `is_on` property. 

Its a mix of both currently. The SSE events are per setting so fit in nicely with the switch model. The client and actual callbacks for SSE are via the coordinator though.

Reply:
Let's see when you implement :)